ote_sdk/ote_sdk/entities/image.py

```diff
@@ -27,6 +31,7 @@ class Image(IMedia2DEntity):
     """
 
     # pylint: disable=too-many-arguments, redefined-builtin
+    @check_input_parameters_type({"file_path": OptionalImageFilePathCheck})
```

Comment:
is it really required that this is a file path to an image?

Reply:
As I see in Image class description - lines 23-30
`class Image(IMedia2DEntity):
    """
    Represents a 2D image.

    The image must be instantiated with either a NumPy array containing the image data
    or a path to an image file.

    :param data: NumPy data.
    **:param file_path: Path to image file**.
    """`
What checks should I add for this input parameter?

Reply:
I used Image class description as expected type of "file_path" parameter lines 22-31:
class Image(IMedia2DEntity):
    """
    Represents a 2D image.

    The image must be instantiated with either a NumPy array containing the image data
    or a path to an image file.

    :param data: NumPy data.
    :**param file_path: Path to image file**.
    """
Should I use other check for file_path parameter?

Reply:
sorry, this question is more for @LeonidBeynenson , I forgot to mention him

@LeonidBeynenson 
Is it required to check whether it is path to an image, folder, etc ? Is not it enough to check whether it is str or not.
I would like to see this `check_input_parameters_type` without any parameter, if it is possible.

Reply:
I think it is required. We need to validate input parameters where it is possible. It is SDL requirement.
IMHO, addition of one parameter into the decorator is not an evil. 🙂

Reply:
@LeonidBeynenson How can you imagine the process when developer will add new functionality and need to understand what is going on with this `check_input_parameters_type` and which parameters should be passed here.